src/util/string.h

```diff
@@ -35,27 +35,40 @@ class StringCollator {
 /// A nullptr-safe variant of the corresponding standard C function.
 ///
 /// Treats nullptr like an empty string and returns 0.
-inline std::size_t strnlen(
+/// The c11 strnlen_s() is not available on all targets
+inline std::size_t strnlen_s(
         const char* str,
-        std::size_t len) {
+        std::size_t maxlen) {
     if (str == nullptr) {
         return 0;
     }
-    // Invoke the global function
-    return ::strnlen(str, len);
+    // Invoke the global function and benefit from SIMD implementations
+    return ::strnlen(str, maxlen);
 }
 
 /// A nullptr-safe variant of the corresponding standard C function.
 ///
 /// Treats nullptr like an empty string and returns 0.
-inline std::size_t wcsnlen(
+/// The c11 wcsnlen_s is not available on all targets
+/// and wcsnlen() is not available on OpenBSD
+inline std::size_t wcsnlen_s(
         const wchar_t* wcs,
-        std::size_t len) {
+        std::size_t maxlen) {
     if (wcs == nullptr) {
         return 0;
     }
-    // Invoke the global function
-    return ::wcsnlen(wcs, len);
+#if !defined(__BSD__) || defined(__USE_XOPEN2K8)
+    // Invoke the global function SIMD implementations
+    return ::wcsnlen(wcs, maxlen);
+#else
+    std::size_t n;
+    for (n = 0; n < maxlen; n++) {
+        if (!wcs[n]) {
+            break;
+        }
+    }
+    return n;
```

Comment:
That would defeat the purpose of `wcsnlen`, i.e. add a safeguard against cases where the buffer does not contain a nullbyte.

Reply:
Ah, I was not aware that this was trying to be a secure implementation. Welp, doesn't really make sense to argue about this anyways since I doubt it will be called often.

Reply:
I did look into where the function is used, but if we are sure that there's always a nullbyte, we don't need to make a `wscnlen` function at all and just use `wcslen` directly. If we aren't sure, we need to check that case.